### DISCUSSION TITLE
Consistently breaking long lists

### DISCUSSION THREAD

**Discussion by schmijos** [prettier/prettier#15247] [Q&A] [answered]
Why would prettier correct this
![CleanShot 2023-08-14 at 09 49 13](https://github.com/prettier/prettier/assets/245443/a0990a6c-08c3-465f-9a48-260ea9f9118d)

to this
![CleanShot 2023-08-14 at 09 49 46](https://github.com/prettier/prettier/assets/245443/efd6c93b-e22f-421e-b4f0-13a61dafe777)

Can I somehow fore prettier to produce more consistency?

**Comment by kachkaev** [comment, marked as answer]
👋 @schmijos! You can add `//` after `[` to force line breaks inside shorter arrays. Alternatively, you can use `// prettier-ignore` before statements in which you don’t like the result of formatting.


## Original behavior

**Prettier 3.0.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBtAOlABFgZhAJwFsARAQxjIDoSBBAFQGV7aA5EgSQFkuB5AIQBStABIAlADSYc+YuUo0GzNpx4C+rBlOx5CpCtTpMW7bn34sA4toC6WALxYAzgAcANgEsY8uAApZ+goEcGRuMACe8mRicLhwBAShPgCUANyYmBg6AVFUlgCitADSAPqFJUYlgtoyerkFxWW0FQwlXDW6cgZ5haXllfSYdo6unt4UfjndAOYhANYAwvEwHrgecAAmtDBpGVBZtV0KDX3NlgCq+QAyVR1Tx71NJRfXbXd13SdPLzeDUMPOdxeHz+D4KWZkRbLVbrDbnWAeNy7KCZaSdQLULgcVj5MSWDj5IznLgVfJFG7VNH3THY3H4wkMYmk8lvKlgmk4vEEokkkhk362ByAsYg6lUIgeKDxabrJYEFZrTbbZEgCQgCAuFbQJzIUBkBIQADuAAV9QgdShQoayOEdWqAEaJMBzOAwRhkIhwK6SuDIXChJxwB1Ol1ulxkMCS6bIGAEACuQZAgYlMfjiacUbccAAinGIPA-QHEwArJwAD0YmZzeYLSH9bkDaoAjjW4MaCBqLSAyE4ALRSzabVUgWNkRFRhYQIhEMjIbtuNzDjNQaZZ7axjz2uPwY3xb1SwsNxMACxgRDcAHVj144K4I3BGOavB4AG5ecJzsBOO0gF8JjhQBsCBumABAeJqtCAYwERZoejYgC4HaBheiQuHOiG3vEL6+mqkqBvK7ZkNMM5wYm4YEPhc72mQ9pwIuaqIZKMAXh4GwwMeyAABwAAxqsELYeMEhHEbOdZFmqlD2ixbEcUgABMapxoG9A0Ra9bwXARC0RsQEbFcZArnGRFwAAYnoFArCuc5kNuEAgAAvvZQA)

```sh
--parser babel
```

**Input:**

```jsx
[
  formData.DATSTANDIMMOBJAHR,
  formData.DATSTANDIMMOBMONAT,
  formData.DATSTANDIMMOBTAG,
] = splitDate(formData.realtyDataReferralDate);

[
  formData.GEAK_EA_DAT_J,
  formData.GEAK_EA_DAT_M,
  formData.GEAK_EA_DAT_T
] = splitDate(formData.geakCertifiedAt);

[
  formData.GEAK_EA_GUEL_J,
  formData.GEAK_EA_GUEL_M,
  formData.GEAK_EA_GUEL_T
] = splitDate(formData.geakCertifiedUntil);

[
  formData.MINERGIEDATUM_DEKL_J,
  formData.MINERGIEDATUM_DEKL_M,
  formData.MINERGIEDATUM_DEKL_T,
] = splitDate(formData.minergieCertifiedAt);

```

**Output:**

```jsx
[
  formData.DATSTANDIMMOBJAHR,
  formData.DATSTANDIMMOBMONAT,
  formData.DATSTANDIMMOBTAG,
] = splitDate(formData.realtyDataReferralDate);

[formData.GEAK_EA_DAT_J, formData.GEAK_EA_DAT_M, formData.GEAK_EA_DAT_T] =
  splitDate(formData.geakCertifiedAt);

[formData.GEAK_EA_GUEL_J, formData.GEAK_EA_GUEL_M, formData.GEAK_EA_GUEL_T] =
  splitDate(formData.geakCertifiedUntil);

[
  formData.MINERGIEDATUM_DEKL_J,
  formData.MINERGIEDATUM_DEKL_M,
  formData.MINERGIEDATUM_DEKL_T,
] = splitDate(formData.minergieCertifiedAt);

```

## With `//`

**Prettier 3.0.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBtABAekwHSu9AMwgCcBbAEQEMYqA6CgQQBUBlZxgOQoEkBZPgHkAQgClGACQBKAGjwFi5arQYt2XXgJFDOLOfiKlKNekzYdu-IcI4BxfQF10AXnQBnAA4AbAJYxlcAAUisYqJHBUXjAAnspUUnCEcCQkkQEAlADceHgY2PKGSiZ0tgCijADSAPrlVWZVovoKRnEl5dW19XxNhaH0ZZU1jHUsVcx4Tq6evv40QSGtAOYRANYAwskwPoQ+cAAmjDBZOVB5uAYLxQMdw7YAqqUAMg09lyrXQ1X3T1XdBW-9dqfb7PcZQSbubx+ALBFrFZZUdabba7PZ3WA+LzHKC5LDnZpFFR8HicUpSWw8UpmO58OqlCrPRr-OFEklkilUlg0ukM36vFn0Ymk8mU6m0ij00GOFyQmYwgF0Mg+KDJRa7DYkLY7faHbEgGQgCAeLbQNzIUBUFIQADuAAVLQgzShItaqNEzQaAEapMArOAwVhUMhwR7KuDIQiRNxwL0+v0BjxUMDKxbIGAkACuMZA0aVacz2bcKa8cAAihmIPAI1HswArNwAD1YxbLFarSEjXmjBoAjm24LaSEanSAqG4ALQq-b7fUgdNUTEptYQMhkKjIUdeLyzotQRYlw7pnyejPwW3JUMq6td7MACxgZC8AHVb344J4k3BWI6-D4AG5+NEG5gG4HogH+WY8FAewIAGYAkD4xqMNBrAxCW17diAHhDtGT6pB4G7Ye+yR-uGBrKtGmqDlQixrhh2aJiQlEbp6VCenA24GthyowE+Ph7DAt7IAAHAADAa4R9j44TUbR64djWBq0J6fECUJSAAEwGhm0bMGxTqdphcBkOxewwXsjxUHuGY0XAABiRg0Fse4blQp4QCAAC+nlAA)

```sh
--parser babel
```

**Input:**

```jsx
[ //
  formData.DATSTANDIMMOBJAHR,
  formData.DATSTANDIMMOBMONAT,
  formData.DATSTANDIMMOBTAG,
] = splitDate(formData.realtyDataReferralDate);

[ //
  formData.GEAK_EA_DAT_J,
  formData.GEAK_EA_DAT_M,
  formData.GEAK_EA_DAT_T
] = splitDate(formData.geakCertifiedAt);

[ //
  formData.GEAK_EA_GUEL_J,
  formData.GEAK_EA_GUEL_M,
  formData.GEAK_EA_GUEL_T
] = splitDate(formData.geakCertifiedUntil);

[ //
  formData.MINERGIEDATUM_DEKL_J,
  formData.MINERGIEDATUM_DEKL_M,
  formData.MINERGIEDATUM_DEKL_T,
] = splitDate(formData.minergieCertifiedAt);

```

**Output:**

```jsx
[
  //
  formData.DATSTANDIMMOBJAHR,
  formData.DATSTANDIMMOBMONAT,
  formData.DATSTANDIMMOBTAG,
] = splitDate(formData.realtyDataReferralDate);

[
  //
  formData.GEAK_EA_DAT_J,
  formData.GEAK_EA_DAT_M,
  formData.GEAK_EA_DAT_T,
] = splitDate(formData.geakCertifiedAt);

[
  //
  formData.GEAK_EA_GUEL_J,
  formData.GEAK_EA_GUEL_M,
  formData.GEAK_EA_GUEL_T,
] = splitDate(formData.geakCertifiedUntil);

[
  //
  formData.MINERGIEDATUM_DEKL_J,
  formData.MINERGIEDATUM_DEKL_M,
  formData.MINERGIEDATUM_DEKL_T,
] = splitDate(formData.minergieCertifiedAt);

```

## With `prettier-ignore`

**Prettier 3.0.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBtAOlABFgZhAJwFsARAQxjIDoSBBAFQGV7aA5EgSQFkuB5AIQBStABIAlADSYc+YuUo0GzNpx4C+rBlOx5CpCtTpMW7bn34sA4toC6WALxYAzgAcANgEsY8uAApZ+goEcGRuMACe8mRicLhwBAShPgCUANyYmAD0mVguwTAwHvEAtB4A5lCEcJioWNK6cgZUlgCitADSAPptnUadgtoyelHNbV09fVyDDYHUrR3dtL0MnfSYdo6unt4UfgEjZSEA1gDC8YW4RQAmtDBpGVDZufmFJeWVwTX1+03z40uWACqLQAMv1pj8FH9Fp0gaDOlNvsNfmMYXCwWsoBtnO4vD5-MiFIcyKdzh5LnAroDYB43PcoF8dJDqFwOKwWmJLBwWkZAVxei12mCBkjGgpWezOdzefySIKwYimYSWWyOVyeQw+QKhatbA4cdt8cyqEQPFB4mUimcCBdrrd6SAJCAIC5CtAnMhQGQEhAAO4ABW9CA9KFCvrI4Q9ToARokwEc4DBGGQiHAQWa4MhcKEnHAY3GE0mXGQwGaysgYAQAK55kC500V6u1pxltxwACKVYg8CzOdrACsnAAPRitjtdntIbNuXNOgCOE7g-oILpDIDITmK5splMdIErZFpZZOECIRDIyHXbjce5bUDKbdulY80ar8H98XT5t7M9rAAsYCINwAHU-y8OBXBLOBGGDLwPAANy8cJLzAJwoxAeCaw4KArgQJMwAIDxXVoHDGAiNsf1nEA8ggXNgMSFxLzyCD4ngzMnTNXMbWXMgynPSja2LAguMvaMyGjOAbydPIzRgYCPCuGA-2QAAOAAGJ1ggXDxgh4viLynPsnUoaN5MU5SkAAJidKtc3ocSQ2nKi4CICSrlwq4QTIe8q14uAADE9AoQp70vMg3wgEAAF8oqAA)

```sh
--parser babel
```

**Input:**

```jsx
[
  formData.DATSTANDIMMOBJAHR,
  formData.DATSTANDIMMOBMONAT,
  formData.DATSTANDIMMOBTAG,
] = splitDate(formData.realtyDataReferralDate);

// prettier-ignore
[ 
  formData.GEAK_EA_DAT_J,
  formData.GEAK_EA_DAT_M,
  formData.GEAK_EA_DAT_T
] = splitDate(formData.geakCertifiedAt);

// prettier-ignore
[
  formData.GEAK_EA_GUEL_J,
  formData.GEAK_EA_GUEL_M,
  formData.GEAK_EA_GUEL_T
] = splitDate(formData.geakCertifiedUntil);

[
  formData.MINERGIEDATUM_DEKL_J,
  formData.MINERGIEDATUM_DEKL_M,
  formData.MINERGIEDATUM_DEKL_T,
] = splitDate(formData.minergieCertifiedAt);

```

**Output:**

```jsx
[
  formData.DATSTANDIMMOBJAHR,
  formData.DATSTANDIMMOBMONAT,
  formData.DATSTANDIMMOBTAG,
] = splitDate(formData.realtyDataReferralDate);

// prettier-ignore
[ 
  formData.GEAK_EA_DAT_J,
  formData.GEAK_EA_DAT_M,
  formData.GEAK_EA_DAT_T
] = splitDate(formData.geakCertifiedAt);

// prettier-ignore
[
  formData.GEAK_EA_GUEL_J,
  formData.GEAK_EA_GUEL_M,
  formData.GEAK_EA_GUEL_T
] = splitDate(formData.geakCertifiedUntil);

[
  formData.MINERGIEDATUM_DEKL_J,
  formData.MINERGIEDATUM_DEKL_M,
  formData.MINERGIEDATUM_DEKL_T,
] = splitDate(formData.minergieCertifiedAt);

```

Hope it helps!

**Reply by schmijos** [reply]
Nice, thank you!
